src/H5FD.c

```diff
@@ -104,16 +111,93 @@ H5FD_init(void)
     herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_NOAPI(FAIL)
+    /* FUNC_ENTER() does all the work */
+
+done:
+    FUNC_LEAVE_NOAPI(ret_value)
+} /* end H5FD_init() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5FD__init_package
+ *
+ * Purpose:     Initialize the virtual file layer.
+ *
+ * Return:      SUCCEED/FAIL
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5FD__init_package(void)
+{
+    char  *lock_env_var = NULL;    /* Environment variable pointer */
+    herr_t ret_value    = SUCCEED; /* Return value */
+
+    FUNC_ENTER_PACKAGE
 
     if (H5I_register_type(H5I_VFL_CLS) < 0)
         HGOTO_ERROR(H5E_VFL, H5E_CANTINIT, FAIL, "unable to initialize interface");
 
     /* Reset the file serial numbers */
     H5FD_file_serial_no_g = 0;
 
+    /* Check the use disabled file locks environment variable */
+    lock_env_var = getenv(HDF5_USE_FILE_LOCKING);
+    if (lock_env_var && !strcmp(lock_env_var, "BEST_EFFORT"))
+        H5FD_ignore_disabled_file_locks_p = true; /* Override: Ignore disabled locks */
+    else if (lock_env_var && (!strcmp(lock_env_var, "TRUE") || !strcmp(lock_env_var, "1")))
+        H5FD_ignore_disabled_file_locks_p = false; /* Override: Don't ignore disabled locks */
+    else
+        H5FD_ignore_disabled_file_locks_p = FAIL; /* Environment variable not set, or not set correctly */
+
+    /* Initialize all internal VFD drivers, so their driver IDs are set up */
```

Comment:
These calls just register the VFD plugin (and unregister, for the term call below), the actual initialization of the plugin is done lazily as needed, within each plugin.